### DISCUSSION TITLE
Can't get absolute path of json file in `data` folder

### DISCUSSION THREAD

**Discussion by IroncladDev** [vercel/community#239] [Help] [answered]
My folder structure looks something like this:

Root
 | pages
...
 | scripts
---| util.js
 | Data
---| data.json
---| schemas.json
 | public
...

Within scripts, there is a function that edits a JSON file

```javascript
import editJsonFile from 'edit-json-file'
const DIRNAME = process.cwd()
//get the _absolute_ **absolute** path of data.json file
const data = editJsonFile(process.env.JSON_PATH || (DIRNAME + "/data/data.json"))

export function getJSON(){
  return data.get();
}

export function writeJSON(callback){
  let __data = callback(getJSON());
  let keys = Object.keys(__data);
  let values = Object.values(__data);
  for(var i = 0; i < keys.length; i++){
    data.set(keys[i], values[i]);
  }
  data.save();
  return getJSON();
}
```

The edit-json-file npm package needs the absolute path of a json file to be able to edit it.  On my localhost it works fine, but on vercel I get an error:

`{"errno":-30,"syscall":"open","code":"EROFS","path":"/var/task/data/data.json"}`

Is there any way I can fix this or access the data folder?

**Comment by CatsAreMLG** [comment, marked as answer]
You can access the base URL via the [System Environment Variables](https://vercel.com/docs/concepts/projects/environment-variables#system-environment-variables). However, our file system is not writable, you might want to try having a database to store this file on to be able to edit it.
